### PR TITLE
GenAI Reorg Part 3: Create Dataset resource

### PR DESCRIPTION
**Pull request by brendanshean**
## WHAT
Create `Evidence::Research::GenAI::Dataset` resource

## WHY
Being able to [View & Create Datasets](https://www.notion.so/quill/Internal-Tool-Organizing-Trials-into-Datasets-8dbc188caf194157a2a28395e907ddc0?pvs=4#94f3485a71144b95b246ce13b5142dee) is helpful organizing the different trial runs over StemVaults

## HOW
Create a migration/routes/model for Dataset and connect it to StemVault.  Subsequent PRs will add the actual data to the Dataset object.

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/Internal-Tool-Organizing-Trials-into-Datasets-8dbc188caf194157a2a28395e907ddc0?pvs=4

### What have you done to QA this feature?
(Provide enough detail that a reviewer could assess whether additional QA should be done. For larger projects, additionally use the Engineer Feature Testing Notion template. Review Guidelines if needed: https://www.notion.so/quill/Github-PR-QA-Guidelines-49e99fc965654ceeb8c6249bd9d181d7)

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  YES
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES

### CODE REVIEW

File: services/QuillLMS/db/migrate/20240619191903_create_evidence_research_gen_ai_datasets.evidence.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+# This migration comes from evidence (originally 20240619184956)
+class CreateEvidenceResearchGenAIDatasets < ActiveRecord::Migration[7.0]
+  def change
+    create_table :evidence_research_gen_ai_datasets do |t|
+      t.integer :stem_vault_id, null: false
+      t.integer :num_optimal, null: false
```
Comment (dandrabik) [inline services/QuillLMS/db/migrate/20240619191903_create_evidence_research_gen_ai_datasets.evidence.rb]: Maybe let's use `optimal_count` and `suboptimal_count`. We don't typically use abbreviations like `num` in names.

File: services/QuillLMS/engines/evidence/app/views/evidence/research/gen_ai/stem_vaults/show.html.erb [hunk 2]
```diff
@@ -9,27 +9,40 @@
...
+  </h3>
   <ul>
     <% @stem_vault.guidelines.optimal.each do |guideline| %>
       <li><%= guideline %></li>
     <% end %>
   </ul>
 
-  <h3>Sub-Optimal Guidelines</h3>
+  <h3>
+    Sub-Optimal Guidelines
```
Comment (dandrabik) [inline services/QuillLMS/engines/evidence/app/views/evidence/research/gen_ai/stem_vaults/show.html.erb]: I believe `Suboptimal` is [a word](https://www.merriam-webster.com/dictionary/suboptimal#:~:text=%3A%20less%20than%20optimal%20%3A%20not%20at,suboptimal%20conditions%2Fresults%2Fsolutions).

**Review by dandrabik — APPROVED** [review]
Small naming change requests, but I don't need to re-review. Thanks for moving this along!

File: services/QuillLMS/engines/evidence/app/views/evidence/research/gen_ai/stem_vaults/show.html.erb (hunk 2 above)
Comment (brendanshean) [reply services/QuillLMS/engines/evidence/app/views/evidence/research/gen_ai/stem_vaults/show.html.erb]: Cool.  That will be much easier in code than having to switch between sub-optimal and sub_optimal

**Pull request merged**